lib/link/inventory.rb

```diff
@@ -0,0 +1,24 @@
+module Link
+  class Inventory
+    attr_reader :inventory, :user
+
+    delegate :network_partner?, to: :user
+    delegate :facilitator?, :participant?, to: :inventory
+
+    def initialize(inventory, user)
+      @inventory = inventory
+      @user = user
+    end
+
+    def execute
+      target.new(inventory, user).execute
+    end
+
+    private
+    def target
+      return Link::InventoryFacilitator if facilitator?(user: user)
+      return Link::InventoryParticipant if participant?(user: user)
+      Link::Partner
+    end
+  end
+end
```

Comment:
We should capture this in a cleanup task so that we can do this.  Yes, this should ideally be a service, and we should move all of them over at once. :smile: